services/checks/checks.go

```diff
@@ -200,38 +215,32 @@ func (s *Service) executeMySQLChecks(ctx context.Context, checks []check.Check)
 	}
 
 	for _, target := range targets {
-		r, err := models.CreateActionResult(s.db.Querier, target.agentID)
+		r, err := models.CreateActionResult(s.db.Querier, target.agent.AgentID)
 		if err != nil {
-			s.l.Errorf("Failed to prepare action result for agent %s: %s.", target.agentID, err)
+			s.l.Errorf("Failed to prepare action result for agent %s: %s.", target.agent.AgentID, err)
 			continue
 		}
 
 		for _, c := range checks {
```

Comment:
What I don't particularly like in that code that there we have two sets of checks: one is passed as an explicit parameter, the other is available as implicit parameters s.checks. I think we should get rid of `s.checks` completely and always values as parameters.

Reply:
The idea with `s.checks` is that when for some reason we can't download new checks form server we still able to run previously downloaded stuff.